tfhe/src/typed_api/integers/types/base.rs

```diff
@@ -0,0 +1,794 @@
+                    &mut self.ciphertext.borrow_mut(),
+                    &mut borrowed.ciphertext.borrow_mut(),
+                )
+            }
+        });
+
+        GenericInteger::new(inner_result, self.id)
+    }
+
+    fn le(&self, other: B) -> Self::Output {
+        let inner_result = self.id.with_unwrapped_global(|server_key| {
+            let borrowed = other.borrow();
+            if std::ptr::eq(self, borrowed) {
+                let cloned = borrowed.clone();
+                let r = server_key.inner.smart_le(
+                    &mut self.ciphertext.borrow_mut(),
+                    &mut cloned.ciphertext.borrow_mut(),
+                );
+                r
+            } else {
+                server_key.inner.smart_le(
+                    &mut self.ciphertext.borrow_mut(),
+                    &mut borrowed.ciphertext.borrow_mut(),
+                )
+            }
+        });
+
+        GenericInteger::new(inner_result, self.id)
+    }
+
+    fn gt(&self, other: B) -> Self::Output {
+        let inner_result = self.id.with_unwrapped_global(|server_key| {
+            let borrowed = other.borrow();
+            if std::ptr::eq(self, borrowed) {
+                let cloned = borrowed.clone();
+                let r = server_key.inner.smart_gt(
+                    &mut self.ciphertext.borrow_mut(),
+                    &mut cloned.ciphertext.borrow_mut(),
+                );
+                r
+            } else {
+                server_key.inner.smart_gt(
+                    &mut self.ciphertext.borrow_mut(),
+                    &mut borrowed.ciphertext.borrow_mut(),
+                )
+            }
+        });
+
+        GenericInteger::new(inner_result, self.id)
+    }
+
+    fn ge(&self, other: B) -> Self::Output {
+        let inner_result = self.id.with_unwrapped_global(|server_key| {
+            let borrowed = other.borrow();
+            if std::ptr::eq(self, borrowed) {
+                let cloned = borrowed.clone();
+                let r = server_key.inner.smart_ge(
+                    &mut self.ciphertext.borrow_mut(),
+                    &mut cloned.ciphertext.borrow_mut(),
+                );
+                r
+            } else {
+                server_key.inner.smart_ge(
+                    &mut self.ciphertext.borrow_mut(),
+                    &mut borrowed.ciphertext.borrow_mut(),
+                )
+            }
+        });
+
+        GenericInteger::new(inner_result, self.id)
+    }
+}
+
+// This extra trait is needed as otherwise
+//
+// impl<P> FheBootstrap for GenericInteger<P>
+//     where P: StaticCrtParameters,
+//           P: IntegerParameter<InnerCiphertext=CrtCiphertext>,
+// { /* sutff */ }
+//
+// impl<P> FheBootstrap for GenericInteger<P>
+//     where P: StaticRadixParameters,
+//         P: IntegerParameter<InnerCiphertext=RadixCiphertext>,
+//       P::Id: WithGlobalKey<Key=GenericIntegerServerKey<P>>,
+// { /* sutff */ }
+//
+// Leads to errors about conflicting impl
```

Comment:
missing a / or is it a comment ?

Reply:
its a comment explaining why we have to do something more complex that what one could think at first

Reply:
ok !